### PR TITLE
add ansible-sts test tool for HA and fencing verification 

### PR DESCRIPTION
**Pull request by udis**

### CODE REVIEW

File: ansible/osp_sts.yaml [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+        oc cp -n openstack {{ sts_working_dir }}/{{ item }} sts:{{ sts_home_dir }}/{{ item }}
+      environment:
+        <<: *oc_env
+      with_items:
+      - ansible_ha.env
+      - ansible.cfg
+      - hosts
+      - ssh.config.ansible
+
+    - name: Copy hosts and ssh key file from openstackclient pod to localhost
```
Comment (pinikomarov) [inline ansible/osp_sts.yaml:139]: you would benefit from a controller hostname -s check on all controllers command 
to see if the setup phase of the tests has gone well

File: ansible/osp_sts.yaml [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        oc exec -n openstack sts -- bash -c "cp -av {{ sts_operator_test_dir }}/* {{ sts_home_dir }}"
+      environment:
+        <<: *oc_env
+
+    - name: Install rhos-release and ansible
+      shell: |
+        #!/bin/bash
+        set -e
+        oc rsh -n openstack sts sudo yum install -y http://download.devel.redhat.com/rcm-guest/puddles/OpenStack/rhos-release/rhos-release-latest.noarch.rpm
+        oc rsh -n openstack sts sudo rhos-release -O 16.2
```
Comment (pinikomarov) [inline ansible/osp_sts.yaml:198]: please add a variable for this , that will make is easier for future usage

Comment (pinikomarov) [reply ansible/osp_sts.yaml:198]: in dev tools I see they use this :
ansible/templates/osp/rhsm.yaml.j2:
16 ## float() returns 0.0 if var cannot converted to float, all upstream release names returns 0.0
17: {% elif osp.release|float() >= 16.2 %}
18 - name: install rhos-release

File: ansible/osp_sts.yaml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+      shell: |
+        #!/bin/bash
+        set -e
+        oc exec -n openstack sts -- bash -c "pushd "{{ sts_home_dir }}/" ; sudo -H -u cloud-admin bash -c 'source ansible_ha.env && timeout {{ osp.deploy_timeout }} sh osp_operator_sanity_run.sh'" > {{ working_log_dir }}/sts.log 2>&1
+      environment:
+        <<: *oc_env
+      ignore_errors: true
+      register: sts_run
+
+    - name: Get sts summary
```
Comment (pinikomarov) [inline ansible/osp_sts.yaml:213]: run junit.sh here

**Review by pinikomarov — COMMENTED** [review]
Draft LGTM :)

File: ansible/osp_sts.yaml (hunk 2 above)
Comment (udis) [reply ansible/osp_sts.yaml]: done

File: ansible/osp_sts.yaml (hunk 3 above)
Comment (udis) [reply ansible/osp_sts.yaml]: Done

**Pull request closed without merging**